openmc/lattice.py

```diff
@@ -0,0 +1,1901 @@
+                              'outermost ring). This ring should have {2:d} ' \
+                              'elements.'.format(self._id, r,
+                                                 6*(self._num_rings - 1 - r))
+                        raise ValueError(msg)
+
+        else:
+            axial_slice = self._universes
+            # Check the center ring.
+            if len(axial_slice[-1]) != 1:
+                msg = 'HexLattice ID={0:d} has the wrong number of ' \
+                      'elements in the innermost ring.  Only 1 element is ' \
+                      'allowed in the innermost ring.'.format(self._id)
+                raise ValueError(msg)
+
+            # Check the outer rings.
+            for r in range(self._num_rings-1):
+                if len(axial_slice[r]) != 6*(self._num_rings - 1 - r):
+                    msg = 'HexLattice ID={0:d} has the wrong number of ' \
+                          'elements in ring number {1:d} (counting from the '\
+                          'outermost ring).  This ring should have {2:d} ' \
+                          'elements.'.format(self._id, r,
+                                             6*(self._num_rings - 1 - r))
+                    raise ValueError(msg)
+
+    def find_element(self, point):
+        r"""Determine index of lattice element and local coordinates for a point
+
+        Parameters
+        ----------
+        point : Iterable of float
+            Cartesian coordinates of point
+
+        Returns
+        -------
+        3-tuple of int
+            Indices of corresponding lattice element in :math:`(x,\alpha,z)`
+            or :math:`(\alpha,y,z)` bases
+        numpy.ndarray
+            Carestian coordinates of the point in the corresponding lattice
+            element coordinate system
+
+        """
+        # Convert coordinates to skewed bases
+        x = point[0] - self.center[0]
+        y = point[1] - self.center[1]
+        if self._num_axial is None:
+            iz = 1
+        else:
+            z = point[2] - self.center[2]
+            iz = floor(z/self.pitch[1] + 0.5*self.num_axial)
+        if self._orientation == 'x':
+            alpha = y - x*sqrt(3.)
+            i1 = floor(-alpha/(sqrt(3.0) * self.pitch[0]))
+            i2 = floor(y/(sqrt(0.75) * self.pitch[0]))
+        else:
+            alpha = y - x/sqrt(3.)
+            i1 = floor(x/(sqrt(0.75) * self.pitch[0]))
+            i2 = floor(alpha/self.pitch[0])
+        # Check four lattice elements to see which one is closest based on local
+        # coordinates
+        indices = [(i1, i2, iz), (i1 + 1, i2, iz), (i1, i2 + 1, iz),
+                   (i1 + 1, i2 + 1, iz)]
+        d_min = np.inf
+
+        for idx in indices:
+            p = self.get_local_coordinates(point, idx)
+            d = p[0]**2 + p[1]**2
+            if d < d_min:
+                d_min = d
+                idx_min = idx
+                p_min = p
+
+        return idx_min, p_min
```

Comment:
This return statement assumes that the condition of `d < d_min` in the previous lines of code is always satisfied, as both `idx_min` and `p_min` are defined within that conditions code-block.

I would recommend defining and setting a default for `idx_min` and `p_min` before the `for` loop.